ledger/src/blocktree_db.rs

```diff
@@ -580,10 +581,15 @@ impl Database {
         self.backend.write(batch.write_batch)
     }
 
-    // return the approximate size in bytes of the storage directory, or `None` if an error occurs
-    // while reading the directory (directory not found, file deleted, etc)
-    pub fn storage_size(&self) -> Option<u64> {
-        get_size(&self.path).ok()
+    pub fn storage_size(&self) -> Result<u64> {
+        fs_extra::dir::get_size(&self.path).map_err(|error| {
```

Comment:
How about adding:  `FsExtraError(#[from] fs_extra::error::Error),` to `BlocktreeError` and then removing that `map_err`?

Reply:
thank you, that sounds like the magic I'm looking for!

Reply:
got it, thank you! Had to phrase it as `Ok(...?)` because `fs_extra` declares its own type `Result<T>` which shadowed `std::result::Result`.